### PR TITLE
DCS-879 Extend /locations/{locationId} with optional 'includeInactive' query parameter.

### PR DESCRIPTION
**Pull request by bell-pa**

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/LocationResourceIntTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody().getLocationId()).isEqualTo(-31);
+    }
+
+    @Test
+    public void testGetLocation_Bad_query_parameter_Bad_request() {
+
+        final var response = testRestTemplate.exchange(
+            "/api/locations/-31?includeInactive=Meh",
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/LocationResourceIntTest.java:59]: What happens with `...?includeInactive=`?

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/LocationResource.java [hunk 2]
```diff
@@ -95,8 +97,17 @@ public LocationResource(final AuthenticationFacade authenticationFacade, final L
...
     @GetMapping("/{locationId}")
-    public Location getLocation(@PathVariable("locationId") @ApiParam(value = "The location id of location", required = true) final Long locationId) {
-        return locationService.getLocation(locationId);
+    public Location getLocation(
+        @PathVariable("locationId")
+        @ApiParam(value = "The location id of location", required = true)
+        final Long locationId,
+
+        @RequestParam(value="includeInactive", required = false)
+        @Pattern(regexp="Yes", flags = {Flag.CASE_INSENSITIVE})
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/LocationResource.java:106]: An interesting approach!  In other places we have used booleans...  I wonder if we should do the same for consistency?

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/LocationResourceIntTest.java (hunk 1 above)
Comment (bell-pa) [reply src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/LocationResourceIntTest.java]: Added test

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/LocationResource.java (hunk 2 above)
Comment (bell-pa) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/LocationResource.java]: Replaced with Boolean

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**